File: 2020/ex.cpp
#include<iostream>
#include<vector>

#include<cmath>

int main() {

  long long tests;
  std::cin >> tests;

  while (tests--) {
    long long n, h;
    std::cin >> n >> h;

    std::vector<long long> arr(n);
    for (long long i = 0; i < n; i++)
      std::cin >> arr[i];

    const long long size = arr.size();
    long long currMax = arr[0];
    long long sum = 0;
    for (long long idx = 1; idx < size; idx++)
    {
      if ((arr[idx - 1] < arr[idx] && arr[idx] >= currMax) || arr[idx] > currMax)
        currMax = arr[idx];
      sum += 100 * std::max(std::min(h, currMax) - arr[idx], 0ll);
    }
    std::cout << sum;
  }
}

File: 2020/ex.js
const input_z = [2, 2, 4, 2, 3, 5, 6];
const h_z = 100;
const global_xy = 100;
const count = 5;

let currMax = input_z[0];
let sum = 0;
for (let idx = 1; idx < input_z.length; idx++) {
  if (
    (input_z[idx - 1] < input_z[idx] && input_z[idx] >= currMax) ||
    input_z[idx] > currMax
  )
    currMax = input_z[idx];
  sum += global_xy * Math.max(Math.min(h_z, currMax) - input_z[idx], 0);
}

console.log(sum);
